bson/unmarshaling_cases_test.go

```diff
@@ -269,3 +283,35 @@ func (ms *myString) UnmarshalBSON(bytes []byte) error {
 	*ms = myString(s)
 	return nil
 }
+
+type unmarshalCallTracker struct {
+	unmarshalCalled bool
+}
+
+type unmarshalBehaviorTestCase struct {
+	Tracker    unmarshalCallTracker  `bson:"tracker"`
+	PtrTracker *unmarshalCallTracker `bson:"ptr_tracker"`
+}
+
+func (ms *unmarshalCallTracker) UnmarshalBSONValue(bsontype.Type, []byte) error {
```

Comment:
Should we also test this behavior for `UnmarshalBSON`?

Reply:
Good idea!